File: Algorithms/src/2946. Matrix Similarity After Cyclic Shifts/Code/solution.py
class Solution:
    def areSimilar(self, mat: List[List[int]], k: int) -> bool:
        m, n = len(mat), len(mat[0])
        
        k %= n
        
        for i in range(m):
            for j in range(n):
                if i % 2 == 0:
                    new_col = (j + k) % n
                else:
                    new_col = (j - k + n) % n
                
                if mat[i][j] != mat[i][new_col]:
                    return False
        
        return True

File: Algorithms/src/2946. Matrix Similarity After Cyclic Shifts/Code/solution.go
func areSimilar(mat [][]int, k int) bool {
    m := len(mat)
    n := len(mat[0])
    
    k %= n
    
    for i := 0; i < m; i++ {
        for j := 0; j < n; j++ {
            var newCol int
            
            if i%2 == 0 {
                newCol = (j + k) % n
            } else {
                newCol = (j - k + n) % n
            }
            
            if mat[i][j] != mat[i][newCol] {
                return false
            }
        }
    }
    
    return true
}